core/domain/wipeout_service.py

```diff
@@ -1040,14 +1112,14 @@ def _pseudonymize_models_transactional(
             )
         )
 
-        activity_related_models = (
-            activity_related_snapshot_metadata_models +
-            activity_related_snapshot_content_models +
-            [
-                model for model in commit_log_models
-                if getattr(model, commit_log_model_field_name) == activity_id
-            ]
-        )
+        activity_related_models: List[Optional[base_models.BaseModel]] = [
```

Comment:
Why the "optional"? It's not reflected in the list name and I'm wondering why we need it at all.

/cc @vojtechjelinek

Reply:
Optional is removed and assert is added with comment. Basically, we cannot use `if exception` here because we are fetching models with those ids that already belong to existing models. So, if we put an Exception then that exception will never be executed.

If we see line 1124 there we are collecting ids and with these same ids we are fetching models in line 1127. The return type of `get_multi` is List[Optional[]] so that's why we have to use assert here.